odoo/addons/base/models/res_partner.py

```diff
@@ -261,6 +263,12 @@ def _compute_partner_share(self):
         for partner in self:
             partner.partner_share = not partner.user_ids or not any(not user.share for user in partner.user_ids)
 
+    @api.depends('vat')
+    def _compute_same_vat(self):
+        for partner in self:
+            partner.same_vat = partner.vat and self.env['res.partner'].search([('vat', '=', partner.vat)], limit=1) or False
```

Comment:
partner.vat and self.env['res.partner'].search([('vat', '=', partner.vat), ('id', '!=', partner.id)], limit=1) or False